src/08_example/02_handling_connections_concurrently.md

```diff
@@ -0,0 +1,34 @@
+# Handling Connections Concurrently
+
+The problem with our code so far is that `listener.incoming()` is a blocking iterator;
+we can't read a new request from this stream until we're done with the previous one.
+One strategy to work around this is to spawn a new Task to handle each connection in the background:
+```rust
+{{#include ../../examples/08_03_concurrent_tcp_server/src/main.rs:main_func}}
+```
+
+This works because under the hood, the `async_std` executor runs `handle_connection` on a separate thread.
+However, this doesn't completely solve our problem: `listener.incoming()` still blocks the executor.
+Even if connections are handled in separate threads, futures running on the main thread
+are blocked while `listener` waits on incoming connections.
```

Comment:
I'd suggest skipping this section to avoid cognitive load. If you do want to keep it, I'd rephrase it as a caution. Having blocking code on any thread in the runtime is dangerous and runtime-specific. Since async-std supports spawning blocking tasks, it is possible to work around this safely, but that's an executor specific thing that I wouldn't cover right here.